src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Properties/InterceptedProjectProperties/AbstractBuildEventValueProvider.AbstractBuildEventHelper.cs

```diff
@@ -0,0 +1,189 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Build.Construction;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.Properties.InterceptedProjectProperties
+{
+    internal abstract partial class AbstractBuildEventValueProvider
+    {
+        public abstract class AbstractBuildEventHelper
+        {
+            private const string _execTask = "Exec";
+            private const string _command = "Command";
+
+            protected AbstractBuildEventHelper(string buildEvent,
+                   string targetName,
+                   Func<ProjectTargetElement, string> getTarget,
+                   Action<ProjectTargetElement> setTargetDependencies)
+            {
+                BuildEvent = buildEvent;
+                TargetName = targetName;
+                GetTarget = getTarget;
+                SetTargetDependencies = setTargetDependencies;
+            }
+
+            private Func<ProjectTargetElement, string> GetTarget { get; }
+            private Action<ProjectTargetElement> SetTargetDependencies { get; }
+            private string BuildEvent { get; }
+            private string TargetName { get; }
+
+            public string GetProperty(ProjectRootElement projectXml)
+            {
+                // Check if project file already has props in place for this.
+                var result = TryGetFromProperty(projectXml);
+                if (result.success)
+                {
+                    return result.property;
+                }
+
+                // Check if build events can be found in targets
+                return GetFromTargets(projectXml);
+            }
+
+            public void SetProperty(string unevaluatedPropertyValue, ProjectRootElement projectXml)
```

Comment:
Get and Set Property are doing different things. Get does not care if there are more than one Task and always uses the First Task. Set, on the other hand, when trying to remove, looks for the target with just one Exec task. 

This behavior accomadates the common scenario. But we need to handle the other scenario of having more than one task because this behavior silently passes without telling the user something was wrong.